src/main/scala/DemoFortis.scala

```diff
@@ -173,6 +181,66 @@ object DemoFortis {
       }
     }
 
+    if (mode.contains("tadaweb")) {
```

Comment:
I would agree. Let's save that for a separate PR. Depending on how checkpointing configuration looks, the task for adding that might be an ideal one for moving this code around.